tools/src/main/java/com/nedap/archie/rmobjectvalidator/RMObjectValidator.java

```diff
@@ -95,22 +126,35 @@ private void validateAttributes(List<RMObjectValidationMessage> result, CAttribu
         result.addAll(emptyObservationErrors);
 
         if (emptyObservationErrors.isEmpty()) {
+
             result.addAll(RMMultiplicityValidation.validate(attribute, pathSoFar + "/" + rmAttributeName, attributeValue));
-            if (attribute.isSingle()) {
+
+            if(attribute.getChildren() == null || attribute.getChildren().isEmpty()) {
+                //no child CObjects. Cardinality/existence has already been validated. Run default RM validations
+                String query = "/" + rmAttributeName;
+                aPathQuery = queryCache.getApathQuery(query);
+                List<RMObjectWithPath> childRmObjects = aPathQuery.findList(lookup, rmObject);
+                result.addAll(runArchetypeValidations(childRmObjects, pathSoFar + query, null));
+            }
+            else if (attribute.isSingle()) {
                 validateSingleAttribute(result, attribute, rmObject, pathSoFar);
             } else {
+
                 for (CObject childCObject : attribute.getChildren()) {
                     String query = "/" + rmAttributeName + "[" + childCObject.getNodeId() + "]";
                     aPathQuery = queryCache.getApathQuery(query);
                     List<RMObjectWithPath> childRmObjects = aPathQuery.findList(lookup, rmObject);
                     result.addAll(runArchetypeValidations(childRmObjects, pathSoFar + query, childCObject));
+                    //TODO: find all other child RM Objects that don't match with a given node id (eg unconstraint in archetype) and
+                    //run default validations against them!
```

Comment:
Is this TODO intentional?

Reply:
Hmm, good question. Not a common scenario, and not for now, but if any data is present in this attribute that is not constrained by a C_OBJECT, but other objects are actually constrained by a c_object, they will not be validated. This happens when someone added some extra objects. That's technically possible, so the todo could be relevant in the future. I'm not wring the solution now, so I'll leave it here for now :)